typedb_ml/typedb/test/base.py

```diff
@@ -77,7 +77,26 @@ def stop(self):
     def _unpack(self):
         self.__unpacked_dir = tempfile.mkdtemp(prefix='typedb')
         with tarfile.open(self.__distribution_location) as tf:
-            tf.extractall(self.__unpacked_dir)
+            def is_within_directory(directory, target):
+                
+                abs_directory = os.path.abspath(directory)
+                abs_target = os.path.abspath(target)
+            
+                prefix = os.path.commonprefix([abs_directory, abs_target])
+                
+                return prefix == abs_directory
+            
+            def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+            
+                for member in tar.getmembers():
+                    member_path = os.path.join(path, member.name)
+                    if not is_within_directory(path, member_path):
+                        raise Exception("Attempted Path Traversal in Tar File")
+            
+                tar.extractall(path, members, numeric_owner) 
+                
+            
+            safe_extract(tf, self.__unpacked_dir)
```

Comment:
note quite our code style but we will merge it and refactor it in a follow up commit/PR